discretisedfield/field.py

```diff
@@ -2455,7 +2455,14 @@ def grad(self):
             msg = f"Cannot compute gradient for nvdim={self.nvdim} field."
             raise ValueError(msg)
 
-        return self.diff("x") << self.diff("y") << self.diff("z")
+        # Create a list of derivatives for each dimension
+        derivatives = [self.diff(dim) for dim in self.mesh.region.dims]
+
+        result = derivatives[0]
+        for derivative in derivatives[1:]:
```

Comment:
Will this give an error if `self.mesh.region.ndim = 1`?

Reply:
This just gives an empty list if `self.mesh.region.ndim = 1` so the for loop never runs